src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -106,13 +109,16 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
 
             if (_host != null)
             {
-                // Precalculate ASCII bytes for header name
+                // Precalculate ASCII bytes for Host header
                 // Note that if _host is null, this is a (non-tunneled) proxy connection, and we can't cache the hostname.
-                // CONSIDER: Cache more than just host name -- port, header name, etc
+                string hostHeader =
+                    ((_sslOptions == null) ? _port != DefaultHttpPort : _port != DefaultHttpsPort) ?
```

Comment:
Nit: this is three comparisons, but it could be two if it were:
```C#
_port != (_sslOptions == null ? DefaultHttpPort : DefaultHttpsPort)
```

Reply:
That does seem clearer.  I pushed an update.